clang/test/CIR/CodeGen/asm.c

```diff
@@ -1,32 +1,316 @@
 // RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fclangir-enable -emit-cir %s -o %t.cir
 // RUN: FileCheck --input-file=%t.cir %s
 
-//CHECK: cir.asm(x86_att, {"" "~{dirflag},~{fpsr},~{flags}"}) operand_attrs = [#cir.optnone] side_effects  : () -> ()
+
+// CHECK: cir.asm(x86_att, operands = [] attrs = []
+// CHECK:        {"" "~{dirflag},~{fpsr},~{flags}"})
 void empty1() {
   __asm__ volatile("" : : : );
 }
 
-//CHECK: cir.asm(x86_att, {"xyz" "~{dirflag},~{fpsr},~{flags}"}) operand_attrs = [#cir.optnone] side_effects  : () -> ()
+// CHECK: cir.asm(x86_att, operands = [] attrs = []
+// CHECK:         {"xyz" "~{dirflag},~{fpsr},~{flags}"}) side_effects
 void empty2() {
   __asm__ volatile("xyz" : : : );
 }
 
-//CHECK: cir.asm(x86_att, {"" "=*m,*m,~{dirflag},~{fpsr},~{flags}"}) operand_attrs = [#cir.optnone, !s32i, !s32i] side_effects %0, %0 : (!cir.ptr<!s32i>, !cir.ptr<!s32i>) -> ()
-void t1(int x) {
+// CHECK: cir.asm(x86_att, operands = [%0, %0 : !cir.ptr<!s32i>, !cir.ptr<!s32i>] attrs = [!s32i, !s32i]
+// CHECK:        {"" "=*m,*m,~{dirflag},~{fpsr},~{flags}"}) side_effects
```

Comment:
Can you explain to me what's going on here for every piece of this operation?

I didn't look at this for couple weeks and now I can't understand what's going on, maybe this is telling something about making this easier to read. I can't make a connection between `operands` and `attrs`, or why in the operands list we have a `%0` without a type, then we have another one with a type, etc - not to mention the number of elements seem arbitrary and hard to correlate back. What are the types in `attrs` supposed to give? Can't they be inferred?

It's very hard to read these operations and get the meaning out of it, perhaps inline asm is just that hard, but I rather we be verbose and make it easier to understand.